addons/product/models/product_template.py

```diff
@@ -538,7 +538,37 @@ def copy(self, default=None):
             default = {}
         if 'name' not in default:
             default['name'] = _("%s (copy)", self.name)
-        return super(ProductTemplate, self).copy(default=default)
+
+        res = super(ProductTemplate, self).copy(default=default)
+
+        # Since we don't copy the attribute line directly, we need to match the extra price and
+        # exclusion of the newly created attribute line values with the old ones.
+        # This matching expects this extra price/exclusion to be uniquely determined by the attribute value e.g. if
+        # two lines uses the same attribute value, the copied extra price will be the same on both lines.
+
+        # price_extra match
+        template_value_by_value = {
+            template_value.product_attribute_value_id: template_value for template_value in
+            self.product_variant_ids.product_template_attribute_value_ids
+        }
+        for template_value in res.product_variant_ids.product_template_attribute_value_ids:
+            match = template_value_by_value.get(template_value.product_attribute_value_id, False)
+            template_value.price_extra = match.price_extra if match else 0
+
+        # exclusion match
+        old_to_new = {
+            template_value_by_value[template_value.product_attribute_value_id]: template_value for template_value in
+            res.product_variant_ids.product_template_attribute_value_ids
+            if template_value.product_attribute_value_id in template_value_by_value
+        }
+        for old, new in old_to_new.items():
+            new.exclude_for = new.exclude_for.create([{
+                'product_template_attribute_value_id': new.id,
+                'product_tmpl_id': res.id,
+                'value_ids': [old_to_new[template_value].id for template_value in exclusion.value_ids],
+            } for exclusion in old.exclude_for])
```

Comment:
Same here, this could be simplified, following the same logic as the comment just above, in the same `ptav in ...`, with something like:
```suggestion
            for excl in ptav.exclude_for:
                copied_ptav.exclude_for = [Command.link(new_exclusion)]
```
Where you keep the `product_tmpl_id` with something like: `res.id if excl.product_tmpl_id == self else excl.product_tmpl_id.id`

Reply:
My mistake, `Command.create` would be better here =]

Reply:
I have refactored the code accordingly 😊